GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -76,42 +76,49 @@ public void Execute()
             string mountError = null;
             Version newVersion = null;
 
-            if (this.TryInitialize(out error))
+            try
             {
-                try
+                if (this.TryInitialize(out error))
                 {
-                    if (!this.TryRunUpgrade(out newVersion, out error))
+                    try
                     {
-                        this.ExitCode = ReturnCode.GenericError;
+                        if (!this.TryRunUpgrade(out newVersion, out error))
+                        {
+                            this.ExitCode = ReturnCode.GenericError;
+                        }
+                    }
+                    finally
+                    {
+                        if (!this.TryMountRepositories(out mountError))
+                        {
+                            mountError = Environment.NewLine + "WARNING: " + mountError;
+                            this.output.WriteLine(mountError);
+                        }
+
+                        this.DeletedDownloadedAssets();
                     }
                 }
-                finally
+                else
                 {
-                    if (!this.TryMountRepositories(out mountError))
+                    this.ExitCode = ReturnCode.GenericError;
+                }
+
+                if (this.ExitCode == ReturnCode.GenericError)
+                {
+                    error = Environment.NewLine + "ERROR: " + error;
+                    this.output.WriteLine(error);
+                }
+                else
+                {
+                    if (newVersion != null)

```

Comment:
I can see why this might be a bit hard to follow - 

1) We are using the presence of `newVersion` to indicate whether upgrade was run or not
2) The string that is generated does included an embedded ternary operator that takes into account the `mountError`

However, the actual change being made here is to encapsulate this logic in a try/finally block to run dispose (you can confirm by ignoring whitespace differences). I think we should make these suggested cleanups in a separate PR (I can submit one that can be handled separately...).